pmtiles_file.cpp

```diff
@@ -67,6 +67,34 @@ std::vector<pmtiles::entry_zxy> pmtiles_entries_tms(const char *pmtiles_map, int
 	return filtered;
 }
 
+struct zxycmp {
+	bool operator()(const pmtiles::entry_zxy &a, const pmtiles::entry_zxy &b) {
+		if (a.z < b.z) {
+			return true;
+		} else if (a.z == b.z) {
+			if (a.x < b.x) {
+				return true;
+			} else if (a.x == b.x) {
+				// reversed because the mbtiles SELECT
+				// orders by tile_row, which is TMS
+				if (a.y > b.y) {
+					return true;
```

Comment:
This matches https://github.com/protomaps/PMTiles/blob/main/cpp/pmtiles.hpp#L352 - my intention was for the `pmtiles_entries_tms` to return in exactly TMS order which should match the MBTiles order used in decode. Is there another reason we need to sort again in here?

Reply:
You're right, this actually isn't necessary, and I don't know why I thought it was. I'll revert this part of the change.